app/src/components/Tabs/my-profile-tab.tsx

```diff
@@ -111,38 +173,31 @@ const MyProfileTab: FC<MyProfileTabProps> = ({ session, status, t, lng }) => {
 
   const [searchTerm, setSearchTerm] = useState<string>("");
   const [role, setRole] = useState<string>("");
-  const [filteredUsers, setFilteredUsers] = useState<Array<UserDetails>>([]);
+  const [filteredUsers, setFilteredUsers] = useState<Array<UserAttributes>>([]);
   const [filteredUsersByRole, setFilteredUsersByRole] = useState<
-    Array<UserDetails>
+    Array<UserAttributes>
   >([]);
 
   useEffect(() => {
-    const users = [
-      { id: "1", name: "John Doe", email: "john@example.com", role: "admin" },
-      {
-        id: "2",
-        name: "Jane Smith",
-        email: "jane@example.com",
-        role: "contributor",
-      },
-    ];
-
-    const result = users.filter(
-      (users) =>
-        users.name
-          .toLocaleLowerCase()
-          .includes(searchTerm.toLocaleLowerCase()) ||
-        users.email
-          .toLocaleLowerCase()
-          .includes(searchTerm.toLocaleLowerCase()),
-    );
+    if (cityUsers) {
+      const result = cityUsers.filter(
+        (users: any) =>
+          users.name
+            .toLocaleLowerCase()
+            .includes(searchTerm.toLocaleLowerCase()) ||
+          users.email
+            .toLocaleLowerCase()
+            .includes(searchTerm.toLocaleLowerCase()),
+      );
 
-    setFilteredUsers(result);
-  }, [role, searchTerm]);
+      setFilteredUsers(result);
+    }
+  }, [role, searchTerm, cityUsers]);
 
   useEffect(() => {
-    const selectedUserByRole = filteredUsers.filter((users) =>
-      users.role.toLocaleLowerCase().includes(role.toLocaleLowerCase()),
+    const selectedUserByRole = filteredUsers.filter(
```

Comment:
You're returning multiple users here.
```suggestion
    const selectedUsersByRole = filteredUsers.filter(
```